framework/src/meshgenerators/ParsedExtraIDMeshGenerator.C

```diff
@@ -0,0 +1,58 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ParsedExtraIDMeshGenerator.h"
+#include "libmesh/elem.h"
+#include "ExtraIDUtils.h"
+
+registerMooseObject("MooseApp", ParsedExtraIDMeshGenerator);
+
+InputParameters
+ParsedExtraIDMeshGenerator::validParams()
```

Comment:
We need to rename this object. The name tells me very little about what it does. "parsed" has a very specific definition in moose that I do not believe fits this action

Reply:
I will think about a new name of this mesh generator which can represent its functionality. Let me know if you have any suggestion on renaming.

Reply:
What about `UniqueExtraIDMeshGenerator`?